lte/gateway/python/magma/pipelined/rule_mappers.py

```diff
@@ -106,8 +106,9 @@ def update_all_ue_versions(self, imsi: str, ip_addr: IPAddress):
             ip_addr_str = ip_addr.address.decode('utf-8').strip()
         with self._lock:
             for k, v in self._version_by_imsi_and_rule.items():
-                _, imsi, ip_addr_str, _ = SubscriberRuleKey(*json.loads(k))
-                if imsi == encoded_imsi and ip_addr_str == ip_addr_str:
+                _, cur_imsi, cur_ip_addr_str, _ = SubscriberRuleKey(*json.loads(k))
+                if cur_imsi == encoded_imsi and (ip_addr_str == "" or
```

Comment:
I think its better to do `cur_imsi == encoded_imsi and ip_addr_str== cur_ip_addr_str` (we don't want this check to pass when ip_addr_str is "" but cur_ip_addr_str is some ip addr)

Reply:
this function needs to update version for all rules for given IMSI when IP address is not specified.
if the `ip_addr_str == ""` match is dropped it would not match any rule with valid IP address. Am I missing something?

Reply:
it updates all version when rule_id is not given/ip address is assumed to be available